core/src/test/java/com/google/zxing/oned/EAN8WriterTestCase.java

```diff
@@ -42,4 +42,14 @@ public void testAddChecksumAndEncode() throws WriterException {
     assertEquals(testStr, BitMatrixTestCase.matrixToString(result));
   }
 
+  @Test
+  public void testEncodeIllegalCharacters() throws WriterException {
```

Comment:
You should be able to write `@Test(expected = WriterException.class)` and then the body of the test is just the line that causes the exception.